gitlab.go

```diff
@@ -291,8 +294,7 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 
 	response := newResponse(resp)
 
-	err = CheckResponse(resp)
-	if err != nil {
+	if err := CheckResponse(resp); err != nil {
```

Comment:
As for the `[]*Foo` vs `[]Foo` discussion, I also searched the standard packages for uses of this idiom. It's indeed used quite often, but in most cases when no `err` variable already exists it the current scope. In cases where a `err` variable already exists, I notice that they tent to reuse it and not create a new shadow variable.

So again... I do want this package to follow the best Go idioms out there, but I doubt this is really one of them. Any additional info reasoning/explaining/discussing this would be appreciated to learn why you think this is (as you say it) "the preferred style".